tron/core/actionrun.py

```diff
@@ -1314,6 +1314,34 @@ def kill(self, final: bool = True) -> Optional[str]:
 
         return "\n".join(msgs)
 
+    def _exit_unsuccessful(self, exit_status=None, retry_original_command=True) -> Optional[Union[bool, ActionCommand]]:
+
+        k8s_cluster = KubernetesClusterRepository.get_cluster()
+        disable_retries_on_lost = False if not k8s_cluster else k8s_cluster.disable_retries_on_lost
+
+        if self.is_done:
+            log.info(
+                f"{self} got exit code {exit_status} but already in terminal " f'state "{self.state}", not retrying',
+            )
+            return None
+        if self.last_attempt is not None:
+            self.last_attempt.exit(exit_status)
+        if self.retries_remaining is not None:
+            if disable_retries_on_lost and exit_status == exitcode.EXIT_KUBERNETES_TASK_LOST:
+                log.info(f"{self} skipping auto-retries due to disable_retries_on_lost being enabled.")
+            else:
+                if self.retries_remaining > 0:
+                    self.retries_remaining -= 1
+                    return self.restart(original_command=retry_original_command)
+                else:
+                    log.info(
+                        f"Reached maximum number of retries: {len(self.attempts)}",
+                    )
+        if exit_status is None:
+            return self._done("fail_unknown", exit_status)
+        else:
+            return self._done("fail", exit_status)
```

Comment:
Might be easier for us to maintain in the long run to have a condition catch if disable_retries_on_lost & exit_status ==EXIT_KUBERNETES_TASK_LOST and then handle that case as we want, but then depend on super()._exit_unsuccessful so we don't have to recreate ActionRun's retry handling here?

Alternatively, you could add a new param to ActionRun config classes that includes the list of 'non-retryable' exit codes (if any; our default would be None/empty list), and then we'd only need to override that param for KubernetesActionRun (preferably from k8s_config as well), rather than override `_exit_unsuccessful`.  

It means we wouldn't need a special toggle in config just for 'disabling retries on LOST', the toggle would just be setting this list to be `[EXIT_KUBERNETES_TASK_LOST]` instead of `[]` for example

Reply:
editing `super()._exit_unsuccessful` like this is a lot simpler to maintain, but it means that to enable the toggle we'll need to release a new tron version everywhere instead of just changing tron config. 

Is this a behavior that we want???


```python
NON_RETRYABLE_EXIT_CODES: Set[int] = {exitcode.EXIT_KUBERNETES_TASK_LOST}

    def _exit_unsuccessful(self, exit_status=None, retry_original_command=True) -> Optional[Union[bool, ActionCommand]]:
        if self.is_done:
            log.info(
                f"{self} got exit code {exit_status} but already in terminal " f'state "{self.state}", not retrying',
            )
            return None
        if self.last_attempt is not None:
            self.last_attempt.exit(exit_status)
        if self.retries_remaining is not None:
            if exit_status in NON_RETRYABLE_EXIT_CODES:
                log.info(f"{self} skipping auto-retries due to non-retryable exit code was recieved.")
            else:
                if self.retries_remaining > 0:
                    self.retries_remaining -= 1
                    return self.restart(original_command=retry_original_command)
                else:
                    log.info(
                        f"Reached maximum number of retries: {len(self.attempts)}",
                    )
        if exit_status is None:
            return self._done("fail_unknown", exit_status)
        else:
            return self._done("fail", exit_status)
```

Reply:
>  it means that to enable the toggle we'll need to release a new tron version everywhere instead of just changing tron config.

Yup we definitely don't want that!

That's where i was suggesting
> and then we'd only need to override that param for KubernetesActionRun (**preferably from k8s_config as well**)

Or this config doesn't even have to be k8s-specific; we can just have a full list and it can contain any exit codes (meaning we could combine from different executor-specific codes too, if we wanted), then the top level ActionRun would pull from that config.

Reply:
i'm a fan of the non-retryable exit code list - it seems like it'd be more usable past this migration and should hopefully be less code too

Reply:
(we can also use srv-configs for this list - it doesn't necessarily have to be sourced from MASTER.yaml :))